clients/cli/src/list.rs

```diff
@@ -1,44 +1,209 @@
-use lockbook_core::{Core, Error, GetFileByPathError};
-use lockbook_core::{Filter, Uuid};
-
-use crate::error::CliError;
-use crate::utils::print_last_successful_sync;
-
-pub fn list(
-    core: &Core, ids: bool, documents: bool, folders: bool, all: bool,
-) -> Result<(), CliError> {
-    core.get_account()?;
-
-    let file_filter = if documents {
-        Some(Filter::DocumentsOnly)
-    } else if folders {
-        Some(Filter::FoldersOnly)
-    } else if all {
-        None
-    } else {
-        Some(Filter::LeafNodesOnly)
-    };
+use std::cmp::Ordering;
+use std::path::Path;
+
+use clap::Parser;
 
-    let paths = core.list_paths(file_filter)?;
+use lb::Core;
+use lb::File;
+use lb::Uuid;
 
-    for path in paths {
-        if ids {
-            println!("{}: {}", get_by_path(core, &path)?, path)
+use crate::CliError;
+use crate::ID_PREFIX_LEN;
+
+#[derive(Parser, Debug)]
+pub struct ListArgs {
+    /// include all children of the given directory, recursively
+    #[clap(short, long)]
+    recursive: bool,
+    /// include more info (such as the file ID)
+    #[clap(short, long)]
+    long: bool,
```

Comment:
```suggestion
    long: bool,
    
```
and so on?

Reply:
https://github.com/lockbook/lockbook/pull/1561/commits/40ade61b05a12dc9a3dd6c32aa8fe73ecfb399ac